### PR TITLE
Implement Slider on Skills page #2427

### LINKED ISSUE #2427: Implement Slider on Skills page
On Amazon there is a slider above the skills. Please implement the same on the Skills page. Create a slides for:
* Devices, add an image of a Raspi here -> link to https://susi.ai/devices (move "device page" here)
* Android app with a link to Playstore and Fdroid
* Developers: Setup SUSI.AI on Your Machine -> link to https://github.com/fossasia?utf8=✓&q=susi

Compare:
https://www.amazon.de/b?ie=UTF8&language=en_GB&node=10068460031

Screenshot from 2019-07-03 12-42-36

### PR DESCRIPTION
**Pull request by fragm3**
Fixes #2427

Changes: Implement Slider on Skills page

Demo Link: https://pr-2429-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 

Screenshot 2019-07-04 at 2 55 16 AM

Screenshot 2019-07-04 at 2 55 26 AM

Screenshot 2019-07-04 at 2 55 32 AM

Mobile view:

Screenshot 2019-07-04 at 3 04 37 AM

### CODE REVIEW

**Comment by mariobehling** [comment]
Images have different size and I cannot see they move automatically.

File: src/components/cms/BrowseSkill/Carousal.css [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.slick-slide {
```
Comment (akshatnitd) [inline src/components/cms/BrowseSkill/Carousal.css]: We already have slick styles used in ChatApp.css. KIndly remove both and move it to a higher scope level.

File: src/components/cms/BrowseSkill/Carousal.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+function PrevArrow({ onClick }) {
+  return <ChevronLeft onClick={onClick} className="slick-prev" />;
+}
+
+PrevArrow.propTypes = {
+  onClick: PropTypes.func,
+};
+
+const Carousal = ({ history }) => {
```
Comment (akshatnitd) [inline src/components/cms/BrowseSkill/Carousal.js]: Typo `Carousel`

File: src/components/cms/BrowseSkill/Carousal.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    adaptiveHeight: true,
+  };
+  return (
+    <div style={{ margin: '1rem' }}>
+      {/* <Container> */}
+      <Slider {...settings}>
+        <SliderImage
+          src={raspberrypi}
+          onClick={() => history.push('/devices')}
+        />
```
Comment (akshatnitd) [inline src/components/cms/BrowseSkill/Carousal.js]: Store the image and url in a constants file.

Comment (akshatnitd) [reply src/components/cms/BrowseSkill/Carousal.js]: It should be array of objects.

File: src/components/cms/BrowseSkill/Carousal.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const Carousal = ({ history }) => {
+  const settings = {
+    className: 'center',
+    centerMode: true,
+    infinite: true,
+    slidesToShow: 1,
+    speed: 500,
+    nextArrow: <NextArrow />,
+    prevArrow: <PrevArrow />,
+    // autoplay: true,
```
Comment (akshatnitd) [inline src/components/cms/BrowseSkill/Carousal.js]: We need autoplay here.

**Comment by fragm3** [comment]
@akshatnitd I don't think we need a high-level component here as we are using it for different purposes and the CSS is different as well. I could take this common slider approach in my next PR if you want.

**Comment by fragm3** [comment]
Screenshot:

Screenshot 2019-07-04 at 9 25 42 PM

Screenshot 2019-07-04 at 9 25 57 PM

**Comment by mariobehling** [comment]
Fix this, please:
* Available on Android
* Available on Linux
* Match the height and size of tiles of the sliders - same as Amazon
* Compare the arrows < > in Amazon to ours. On Amazon they are black and white and work on dark and brighter background 
Screenshot from 2019-07-04 21-54-25
* Position the arrows on the outside sliders instead of on the center slider (like Amazon)
Screenshot from 2019-07-04 21-58-59

**Comment by mariobehling** [comment]
Please also move the search bar below the slider.

Screenshot from 2019-07-04 22-01-43

**Comment by mariobehling** [comment]
Change sliders as follows:
* Set up on your local machine -> Add a download on Github button
* Available on Android 
  * Is it possible to have two buttons here: Get it on Playstore / Get it on Fdroid
  * Please increase the size of the SUSI.AI screenshot
Screenshot from 2019-07-04 23-50-57

**Comment by fragm3** [comment]
@mariobehling As we are using images, adding two separate links won't be possible.

**Comment by mariobehling** [comment]
> @mariobehling As we are using images, adding two separate links won't be possible.

It could be done with layers. We can do this in a next step. Please finalize the other points and ensure small things like using a capital for the Android name are taken care of. I don't understand why you write it as "android" as a name. Ensure those small things don't waste the time of reviewers, please.

**Comment by fragm3** [comment]
@mariobehling 
Screenshot 2019-07-05 at 9 50 51 PM



I have tried to replicate it as just like Amazon, but `react-slick` follows its own design rules, which didn't allow me to set the width as I wished. If I decrease the width, the slider would create white spaces.

Also, the icon used by Amazon is not present in Material UI, I'll look in a way to create icons dynamically. For this, this PR is good to go.

**Comment by mariobehling** [comment]
Thank you! Please also see a follow up issue here: https://github.com/fossasia/susi.ai/issues/2455

**Comment by mariobehling** [comment]
@fragm3 I see you opened a branch for this PR in the repo itself. Please follow best practices and make PRs from your own fork.

**Pull request merged**